### ISSUE TITLE
Use schema instead of database for shadow database where available

### ISSUE THREAD

**Issue by janpio** [prisma/prisma#8059]
Creating a database is an "expensive" action in that it might even be billed for and often is restricted for users. Creating a new schema on the other hand, is usually considered "cheap" and possible.

**Comment by tomhoule** [comment]
Putting "label/discussion" on this because it wouldn't work in the general case. You can absolutely use migrations to manage multiple schemas, and you can bake the name of your schemas into your migrations. Currently it works, but if we used a temporary schema for the shadow database, it wouldn't work anymore.

**Comment by janpio** [comment]
So the limiting factor here is the fact that schemas do not provide an isolated enough environment to what we allow. 

Is schema usage in a migration / DDL SQL well enough defined so we could detect it and maybe use a schema if we deem it safe enough? The requirement to have a second database (or allow creation of it) is one of the bigger ones currently.

**Comment by tomhoule** [comment]
I don't think we could get around this without parsing and analyzing the SQL. Which is not completely off the table, there are libraries we could use, but we would definitely be signing up for more complexity.
